_episodes_rmd/05-data-visualization.Rmd

```diff
@@ -142,129 +133,163 @@ Building plots with **`ggplot2`** is typically an iterative process. We start by
+
+ggplot(data = variants, aes(x = POS, y = DP)) +
+  geom_point(alpha = 0.5, color = "blue")
 ```
 
-We could also color code each of our points according to the number of
-reads (after filtering) that support that variant. This information
-is in the `DP` column: 
+Or to color each species in the plot differently, you could use a vector as an input to the argument **color**. **`ggplot2`** will provide a different color corresponding to different values in the vector. Here is an example where we color with **`sample_id`**:
+
+```{r color-by-sample-1, purl=FALSE}
+ggplot(data = variants, aes(x = POS, y = DP, color = sample_id)) + 
+  geom_point(alpha = 0.5)
+```
 
+Notice that we can change the geom layer and colors will be still determined by **`sample_id`**
 
-```{r color-by-DP, purl=FALSE}
-ggplot(data = variants, aes(x = POS, y = sample_id))+
-geom_point(aes(color = DP))
+```{r color-by-sample-2, purl=FALSE}
+ggplot(data = variants, aes(x = POS, y = DP, color = sample_id)) + 
+  geom_jitter(alpha = 0.5)
+```
+
+To make our plot more readable, we can add axis labels:
+
+```{r add-axis-labels, purl=FALSE}
+ggplot(data = variants, aes(x = POS, y = DP, color = sample_id)) + 
+  geom_jitter(alpha = 0.5) + labs(x = "Base Pair Position", y = "Read Depth (DP)")
 ```
 
 > ## Challenge
 >
-> Use what you just learned to create a scatter plot of `DP` 
-> versus
-> `sample_id`. Is this a good
-> way to show this type of data?
->
->> ## Solution
->> ```{r scatter-challenge, answer=TRUE, purl=FALSE}
->> ggplot(data = variants, aes(x = DP, y = sample_id)) +
->>    geom_point()
->> ```
+> Use what you just learned to create a scatter plot of mapping quality (`MQ`) over
+> position (`POS`) with the samples showing in different colors. Make sure to give your plot
+> relevant axis labels.
+> 
+> > ## Solution
+> > ```{r scatter-challenge, purl=FALSE}
+> >  ggplot(data = variants, aes(x = POS, y = MQ, color = sample_id)) + 
+> >  geom_point() + labs(x="Base Pair Position", y="Mapping Quality (MQ)")
+> > ```
 > {: .solution}
 {: .challenge}
 
-## Boxplot
+## Faceting
 
-When we have one continuous and one categorical variable
-it is usually much more appropriate to make a boxplot
-than a scatterplot. In this case, our continuous variable
-is read depth (`DP`) and our categorical variable in
-`sample_id`. We can use boxplots to visualize the distribution of (filtered) sequencing depth `DP` within each sample:
+**`ggplot2`** has a special technique called *faceting* that allows the user to split one
+plot into multiple plots based on a factor included in the dataset. We will use it to split our
+mapping quality plot into three panels, one for each sample.
 
-```{r boxplot, purl=FALSE}
-ggplot(data = variants, aes(x = sample_id, y = DP))+
-geom_boxplot()
+```{r first-facet, purl=FALSE}
+ggplot(data = variants, aes(x = POS, y = MQ, color = sample_id)) + 
+ geom_point() + labs(x="Base Pair Position", y="Mapping Quality (MQ)") + 
```

Comment:
Same comment as above, and also have indent of `facet_grid()` match that of the geoms.